pkg/controller/backup_configuration.go

```diff
@@ -0,0 +1,249 @@
+			namespace := backupconfiguration.Namespace
+			resource_name := backupconfiguration.Spec.Target.Ref.Name
+			if err := c.CreateOrDeleteEvent(kind, namespace, resource_name); err != nil {
+				return err
+			}
+
+		}
+	}
+	return nil
+}
+
+func (c *StashController) EnsureV1beta1Sidecar(backupconfiguration *api_v1beta1.BackupConfiguration) error {
+	if backupconfiguration == nil {
+		return fmt.Errorf("BackupConfiguration is nil")
+	}
+	kind := backupconfiguration.Spec.Target.Ref.Kind
+	namespace := backupconfiguration.Namespace
+	resource_name := backupconfiguration.Spec.Target.Ref.Name
+
+	if err := c.CreateOrDeleteEvent(kind, namespace, resource_name); err != nil {
+		return err
+	}
+	return nil
+}
+func (c *StashController) CreateOrDeleteEvent(kind, namespace, resource_name string) error {
+	switch kind {
+	case workload_api.KindDeployment:
+		if resource, err := c.dpLister.Deployments(namespace).Get(resource_name); err == nil {
+			key, err := cache.MetaNamespaceKeyFunc(resource)
+			if err == nil {
+				c.dpQueue.GetQueue().Add(key)
+			}
+			return err
+		}
+	case workload_api.KindDaemonSet:
+		if resource, err := c.dsLister.DaemonSets(namespace).Get(resource_name); err == nil {
+			key, err := cache.MetaNamespaceKeyFunc(resource)
+			if err == nil {
+				c.dsQueue.GetQueue().Add(key)
+			}
+			return err
+		}
+	case workload_api.KindStatefulSet:
+		if resource, err := c.ssLister.StatefulSets(namespace).Get(resource_name); err == nil {
+			key, err := cache.MetaNamespaceKeyFunc(resource)
+			if err == nil {
+				c.ssQueue.GetQueue().Add(key)
+			}
+		}
+	case workload_api.KindReplicationController:
+		if resource, err := c.rcLister.ReplicationControllers(namespace).Get(resource_name); err == nil {
+			key, err := cache.MetaNamespaceKeyFunc(resource)
+			if err == nil {
+				c.rcQueue.GetQueue().Add(key)
+			}
+			return err
+		}
+	case workload_api.KindReplicaSet:
+		if resource, err := c.rsLister.ReplicaSets(namespace).Get(resource_name); err == nil {
+			key, err := cache.MetaNamespaceKeyFunc(resource)
+			if err == nil {
+				c.rsQueue.GetQueue().Add(key)
+			}
+			return err
+		}
+	}
+	return nil
+}
+
+func (c *StashController) EnsureCronJob(backupconfiguration *api_v1beta1.BackupConfiguration) error {
+	if backupconfiguration == nil {
+		return fmt.Errorf("BackupConfiguration is nil")
+	}
+
+	image := docker.Docker{
+		Registry: c.DockerRegistry,
+		Image:    docker.ImageStash,
+		Tag:      c.StashImageTag,
+	}
+
+	meta := metav1.ObjectMeta{
+		Name:      backupconfiguration.Name,
+		Namespace: backupconfiguration.Namespace,
+	}
+	ref, err := reference.GetReference(scheme.Scheme, backupconfiguration)
+	if err != nil {
+		return err
+	}
+	if err := c.ensureCronJobRoleBinding(ref); err != nil {
+		return err
+	}
+	_, _, err = batch_util.CreateOrPatchCronJob(c.kubeClient, meta, func(in *v1beta1.CronJob) *v1beta1.CronJob {
+		// set backup-configuration as cron-job owner
+		in.OwnerReferences = []metav1.OwnerReference{
```

Comment:
Can we use `metav1.GetObjectReference()` here?

Reply:
fixed